include/swoole_timer.h

```diff
@@ -46,62 +48,87 @@ struct swTimer_node {
     swTimerDestructor destructor;
 };
 
-struct swTimer {
+class Timer {
+
+ private:
     /*--------------signal timer--------------*/
-    swReactor *reactor;
+    Reactor *reactor_ = nullptr;
     swHeap *heap;
-    std::unordered_map<long, swTimer_node *> *map;
-    uint32_t num;
+    std::unordered_map<long, TimerNode *> map;
     uint64_t round;
     long _next_id;
     long _current_id;
-    long _next_msec;
     /*---------------event timer--------------*/
     struct timeval base_time;
-    /*---------------system timer-------------*/
-    long last_time;
     /*----------------------------------------*/
-    int (*set)(swTimer *timer, long exec_msec);
-    void (*close)(swTimer *timer);
-};
+    int (*set)(Timer *timer, long exec_msec) = nullptr;
+    void (*close)(Timer *timer) = nullptr;
+
+    bool init_reactor(swReactor *reactor);
+    bool init_system_timer();
+
+ public:
+    long next_msec_;
+
+    Timer();
+    ~Timer();
+    static int now(struct timeval *time);
 
-int swTimer_init(swTimer *timer, long msec);
-void swTimer_reinit(swTimer *timer, swReactor *reactor);
-bool swTimer_del(swTimer *timer, swTimer_node *node);
-void swTimer_free(swTimer *timer);
-int swTimer_select(swTimer *timer);
-int swTimer_now(struct timeval *time);
-
-static sw_inline swTimer_node *swTimer_get(swTimer *timer, long id) {
-    auto it = timer->map->find(id);
-    if (it == timer->map->end()) {
-        return nullptr;
-    } else {
-        return it->second;
+    inline int64_t get_relative_msec() {
+        struct timeval _now;
+        if (now(&_now) < 0) {
+            return SW_ERR;
+        }
+        int64_t msec1 = (_now.tv_sec - base_time.tv_sec) * 1000;
+        int64_t msec2 = (_now.tv_usec - base_time.tv_usec) / 1000;
+        return msec1 + msec2;
     }
-}
 
-static sw_inline swTimer_node *swTimer_get_ex(swTimer *timer, long id, const enum swTimer_type type) {
-    swTimer_node *tnode = swTimer_get(timer, id);
-    return (tnode && tnode->type == type) ? tnode : NULL;
-}
+    inline Reactor *get_reactor() {
+        return reactor_;
+    }
 
-static sw_inline int64_t swTimer_get_relative_msec() {
-    struct timeval now;
-    if (!SwooleTG.timer) {
-        return SW_ERR;
+    bool init();
+    TimerNode *add(long _msec, bool persistent, void *data, const swTimerCallback &callback);
+    bool remove(TimerNode *tnode);
+    void reinit(swReactor *reactor);
+    int select();
+
+    inline TimerNode *get(long id) {
```

Comment:
Maybe it would be better called that:

```cpp
inline TimerNode *get_timer_node(long id)
```

Because we might get something else later on.